### PR TITLE
Refractor v2

### PR DESCRIPTION
**Pull request by nguy8tri**
# Description
This refractor targets a few more things:

1. Installation Script for installing dependencies of FOUND (install.sh)
 a. For Linux: Debian, Ubuntu, Redhat, Fedora, CentOS (if there exists apt-get, yum, or dnf)
 b. For macOS (via brew)
 c. Rejects other OS systems
 d. Besides Ubuntu/Debian (so apt-get) and Oracle (so yum), the other installation commands are a bit a shot in the dark and need extra verification, but they appear to be correct
2. Makefile extra targets
 a. Code Coverage is now avaliable via the target coverage, and website is in build/documentation/coverage/index.html
 b. Doxygen is now avaliable via the target doxygen_generate, and website is in build/documentation/doxygen/index.html
 c. Doxygen is initial, there is no cover page (because right now, we don't really have anything to show perse)
3. Docker container for executing workflows
4. Workflow: Full build, test, lint, documentation
 a. Doxygen Build upon Pull Request and for Mainline
 b. Code Coverage Generated upon Pull Request and for Mainline
 c. Build should be locatable to associate with PRs
5. Workflow: Addition of Artifacts to Description upon PR update
6. Workflow: Cleanup of Doxygen/Coverage from Pull Requests

# Artifacts for PR #6 (DO NOT CHANGE)
- [Coverage Artifact](https://uwcubesat.github.io/found/6/merge/coverage)
- [Doxygen Artifact](https://uwcubesat.github.io/found/6/merge/doxygen)

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        else
+            echo "No known package manager found"
+            exit 1
+        fi
+        INSTALL="$PM install -y"
+        ;;
+    Darwin*)
+        # Check if Homebrew is installed; install it if not
+        if ! command -v brew &> /dev/null; then
+            echo "Homebrew not found. Installing Homebrew..."
```
Comment (zeddie888) [inline install.sh:65]: Maybe it's better not to do this, avoid causing as many problems as we can lol

File: src/spatial/attitude-utils.hpp [hunk 2]
```diff
@@ -185,9 +227,13 @@ class Quaternion {
     bool IsUnit(decimal tolerance) const;
     Quaternion Canonicalize() const;
 
+    /// The real component
     decimal real;
+    /// The i component
```
Comment (zeddie888) [inline src/spatial/attitude-utils.hpp:232]: nit: slightly redundant comment, variable name is already very clear

File: src/spatial/camera.hpp [hunk 3]
```diff
@@ -56,13 +59,36 @@ class Camera {
...
+     * @return The focal length of this
+     */
     decimal FocalLength() const { return focalLength; }
+
     decimal Fov() const;
 
     // Mutator Method for Cameras
 
+    /**
+     * Sets the focal length of this
```
Comment (zeddie888) [inline src/spatial/camera.hpp:88]: Specify units, this has been a cause of confusion for LOST

**Review by zeddie888 — CHANGES_REQUESTED** [review]
remove mac installation instructions for now (since untetsed)

File: src/spatial/attitude-utils.hpp (hunk 2 above)
Comment (nguy8tri) [reply src/spatial/attitude-utils.hpp:232]: For doxygen

File: src/spatial/camera.hpp (hunk 3 above)
Comment (nguy8tri) [reply src/spatial/camera.hpp:88]: Answering by commenting the camera object to be in SI.

File: install.sh (hunk 1 above)
Comment (nguy8tri) [reply install.sh:65]: Commented out

**Review by zeddie888 — APPROVED** [review]
lgtm

**Pull request merged**